action-bump-version/action.yaml

```diff
@@ -0,0 +1,75 @@
+name: Bump patch version
+description: |
+  Bump the current patch version in file VERSION
+  and open a new PR with the changes.
+  file VERSION is expected to the patch number
+  on a single line
+  ...
+
+inputs:
+  current-version:
+    required: true
+    description: the current version to bump
+  patch-version-identifier:
+    default: VERSION_PATCH
+    description: |
+      the name of the variable that sets the patch version in the ./VERSION file
+  branch:
+    required: true
+    description: branch name for the new PR to open
+
+runs:
+  using: "composite"
+  steps:
+    - name: Ensure git is installed
+      shell: bash
+      run: |
+        git --version
+    - name: Ensure gh is installed
+      shell: bash
+      run: |
+        gh --version
+    - name: Ensure VERSION file exists
+      shell: bash
+      run: |
+        if [ ! -f ./VERSION ]; then
+          echo "This action bumps the patch version in the file ./VERSION" 1>&2
+          echo "file ./VERSION not found" 1>&2
+          exit 1
+        fi
+    - name: Get next version
+      id: semver
+      uses: actions-ecosystem/action-bump-semver@v1
+      with:
+        current_version: ${{ inputs.current-version }}
+        level: patch
+    - name: Bump component version
+      shell: bash
+      run: |
+        if ! grep ${{ inputs.patch-version-identifier }} ./VERSION ; then
+          echo "file ./VERSION does not contain ${{ inputs.patch-version-identifier }}"
+          exit 1
+        fi
+
+        ID=${{ inputs.patch-version-identifier }}
+        VERSION=${{ steps.semver.outputs.new_version }}
+
+        # update the version file
+        sed -i "s/^$ID=.\+$/$ID=$VERSION\" ./VERSION
+
+        git checkout -b ${{ inputs.branch }} ${{ github.ref_name }}
+
+        git config --local user.email ${{ github.actor }}@scality.com
+        git config --local user.name ${{ github.actor }}
+
+        git add ./VERSION
+
+        git commit -m "Bump version to ${{ steps.outputs.semver.new_version }}"
+        git push --set-upstream origin ${{ inputs.branch }}
```

Comment:
when running on the not-last branch, this will create a conflict and require manual intervention : which completely defeats the purpose....

That conflict however is quite easy to fix (at minimal by merging with 'theirs' strategy to keep the target branch's version ; or by bumping as well on that branch, to be consistent with bert-e's waterflow and fix versions...) : so should actually automate this as well, aka run the merge and fix conflict, then push both the actual PR branch and the `w/<next branch>/<pr branch>` waterflow branches at the same time.